File: ex07/java/kkw5240.java
package ex07.java;

import java.util.Scanner;

class Ex07 {
    private final static double SQUARE_FEET_TO_SQUARE_METER = 0.09290304;

    public static void main(String[] args) {
        double length = 0, width = 0;

        Scanner scanner = new Scanner(System.in);
        try {
            length = promptLength(scanner);
            width = promptWidth(scanner);

        } catch(Exception e) {
            System.out.println(e.getMessage());
            
        } finally {
            scanner.close();
        }

        double areaSquareFeet = getArea(length, width);
        double areaSquareMeter = convertFeetToMeter(areaSquareFeet);

        System.out.printf("""
                You entered dimensions of %.1f feet by %.1f feet
                The area is
                %.3f square feet
                %.3f square meters.
                %n""", length, width, areaSquareFeet, areaSquareMeter);
    }

    private static double promptLength(Scanner scanner) {
        System.out.print("What is the length of the room in feet? ");
        double length = scanner.nextDouble();

        return length;
    }

    private static double promptWidth(Scanner scanner) {
        System.out.print("What is the width of the room in feet? ");
        double width = scanner.nextDouble();

        return width;
    }

    private static double getArea(double length, double width) {
        return length * width;
    }

    private static double convertFeetToMeter(double areaSquareFeet) {
        return areaSquareFeet * SQUARE_FEET_TO_SQUARE_METER;
    }
}

File: ex07/go/kkw5240.go
package main

import "fmt"

const (
	feetToMeter = 0.09290304
)

func main() {
	length, err := promptLength()
	if err != nil {
		return
	}

	width, err := promptWidth()
	if err != nil {
		return
	}

	fmt.Println(
		buildMessage(length, width),
	)
}

func promptWidth() (float32, error) {
	fmt.Print("What is the width of the room in feet? ")

	var width float32
	_, err := fmt.Scanln(&width)
	if err != nil {
		return 0, err
	}
	return width, nil
}

func promptLength() (float32, error) {
	fmt.Print("What is the length of the room in feet? ")

	var length float32
	_, err := fmt.Scanln(&length)
	if err != nil {
		return 0, err
	}
	return length, err
}

func buildMessage(length float32, width float32) string {
	return fmt.Sprintf(
		"You entered dimensions of %.1f feet by %.1f feet "+
			"\nThe area is %.3f square feet "+
			"\n%.3f square meters.",
		length,
		width,
		calcArea(length, width),
		convertFeetToMeter(
			calcArea(length, width),
		),
	)
}

func calcArea(length float32, width float32) float32 {
	return length * width
}

func convertFeetToMeter(feet float32) float32 {
	return feet * feetToMeter
}
